src/tekton/deploy.ts

```diff
@@ -0,0 +1,72 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+import * as os from 'os';
+import * as path from 'path';
+import { cli } from '../cli';
+import * as fs from 'fs-extra';
+import * as yaml from 'js-yaml';
+import * as vscode from 'vscode';
+import { contextGlobalState } from '../extension';
+import { tektonYaml } from '../yaml-support/tkn-yaml';
+import { pipelineExplorer } from '../pipeline/pipelineExplorer';
+import { getStderrString, Command, newK8sCommand } from '../tkn';
+
+
+function checkDeploy(): boolean {
+  return vscode.workspace
+    .getConfiguration('vs-tekton')
+    .get<boolean>('deploy');
+}
+
+export async function updateTektonResource(document: vscode.TextDocument): Promise<void> {
+  let value: string;
+  if (!checkDeploy()) return;
+  if (document.uri.authority !== 'loadtektonresource') {
+    if (document.languageId !== 'yaml') return;
```

Comment:
I like to move this check before `if (document.uri.authority !== 'loadtektonresource') {`
And I think it should be changed to `if (!document.uri.scheme.startsWith('tekton')){` as with new filesystem provider there are no authority `loadtektonresource`

Reply:
right forgot to change according to new authority name :) for filesystem provider